crates/nu-command/src/commands/filesystem/open.rs

```diff
@@ -154,33 +153,32 @@ fn open(args: CommandArgs) -> Result<ActionStream, ShellError> {
     }
 
     // Normal Streaming operation
-    let with_encoding = if encoding.is_none() {
-        None
-    } else {
+    let with_encoding = if let Some(encoding) = encoding {
         Some(get_encoding(encoding)?)
+    } else {
+        None
     };
 
-    let sob_stream = shell_manager.open(&path.item, path.tag.span, with_encoding)?;
+    let reader = shell_manager.open(&path.item, path.tag.span)?;
+    let reader = BufReader::new(reader);
+    let reader = BufCodecReader::new(reader, with_encoding);
 
-    let final_stream = sob_stream.map(move |x| {
-        // The tag that will used when returning a Value
-        let file_tag = Tag {
-            span: path.tag.span,
-            anchor: Some(AnchorLocation::File(path.to_string_lossy().to_string())),
-        };
+    // The tag that will used when returning a Value
+    let file_tag = Tag {
+        span: path.tag.span,
+        anchor: Some(AnchorLocation::File(path.to_string_lossy().to_string())),
+    };
 
-        match x {
-            Ok(StringOrBinary::String(s)) => {
-                ReturnSuccess::value(UntaggedValue::string(s).into_value(file_tag))
-            }
-            Ok(StringOrBinary::Binary(b)) => {
-                ReturnSuccess::value(UntaggedValue::binary(b).into_value(file_tag))
-            }
-            Err(se) => Err(se),
+    let value = match reader.read_to_end()? {
```

Comment:
If you go with `read_to_end`, you'll need to protect nu against stream inputs like `open /dev/urandom`. Not all "files" on the system have an end.

Here's an example I tried in current nushell that streams okay, but won't with this PR:
```
> open /dev/urandom | each { $it }
```